runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -211,6 +211,15 @@ public VersionLockedObject(Supplier<T> newObjectFn,
         // Next, we just upgrade to a full write lock if the optimistic
         // read fails, since it means that the state of the object was
         // updated.
+        // Note (to remove): This can only tell us it was changed by the current client, even in the case that
```

Comment:
This TODO can be removed. I think the main risk is getting the timestamp too early, but that's fine, it will just slightly increase the abort rate.

Reply:
Well actually now that I see @Maithem comment, this is not an actual concern because the snapshotTimestamp has a lazy getter so even though the updateFunction is applied this timestamp is already computed, in this case are we really able to sync?? if the timestamp should not have been updated?